### DISCUSSION TITLE
Handling IFormFile

### DISCUSSION THREAD

**Discussion by sechabamot** [jasontaylordev/CleanArchitecture#995] [General]
I am interested in finding out how you would handle file storage while still maintaining this lovely clean architecture. Mainly not referencing IFormFile in the Application layer.

My first idea was to wrap IFormFile behavior into the Application layer, which seemed like a bit much. From my own research, it seems the approach is to handle the file in the controller.

Would love to hear your thoughts. @jasontaylordev

**Comment by juliengloria** [comment]
Up :)

**Comment by mrhighstone** [comment]
Maybe a little late to the party but this works for me. Below the implementation of the endpoint, instantiating a command holding a stream the file contents.

```
public async Task<ImportFileResult> ImportFile(ISender sender,
    [FromForm] IFormFile importFile)
{
    if (importFile == null)
        return new ImportFileResult();

    using (var stream = importFile.OpenReadStream())
    {
        var command = new ImportFileCommand()
        {
            Content = stream
        };

        return await sender.Send(command);
    }
}
```
